components/ContactForm/index.tsx

```diff
@@ -67,13 +66,47 @@ export const ContactForm = () => {
     return isValid;
   };
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    formIsValid() ? console.log(formState) : console.log("Form is invalid");
+
+    if (formIsValid()) {
+      const target = e.target as typeof e.target & {
+        name: { value: string };
+        email: { value: string };
+        message: { value: string };
+      };
+
+      const data = {
+        name: target.name.value,
+        email: target.email.value,
+        message: target.message.value,
+      };
+
+      const JSONdata = JSON.stringify(data);
+      const endpoint = "/api/contact-us";
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSONdata,
+      };
+
+      const response = await fetch(endpoint, options);
+      const result = await response.json();
+      alert(result.message);
+    } else {
+      console.log("Form is invalid!");
+    }
   };
 
   return (
-    <form onSubmit={handleSubmit} className="flex flex-col gap-8" action="">
+    <form
+      method="POST"
+      onSubmit={handleSubmit}
+      className="flex flex-col gap-8"
+      action="/api/contact-us"
```

Comment:
You are using `e.preventDefault()` which doesn't allow the default native `<form>` behaviour to run. Instead, you're using the `fetch` API inside the `handleSubmit` handler, so you don't need the `method` and `action` attributes. They're only useful if you want the browser to send the request to the endpoint you indicate in `action` with the method you indicate in `method` (the browser just automatically adds the fields from HTML into the request body, using the elements' `name` attribute as a key). This can be a bit confusing, we can discuss on a call if you want.